extmod/modbluetooth.c

```diff
@@ -861,7 +861,10 @@ STATIC mp_obj_t bluetooth_ble_invoke_irq(mp_obj_t none_in) {
         } else if (event == MP_BLUETOOTH_IRQ_GATTC_DESCRIPTOR_RESULT) {
             // conn_handle, handle, uuid
             ringbuf_extract(&o->ringbuf, data_tuple, 2, 0, NULL, 0, &o->irq_data_uuid, NULL);
-        } else if (event == MP_BLUETOOTH_IRQ_GATTC_READ_RESULT || event == MP_BLUETOOTH_IRQ_GATTC_NOTIFY || event == MP_BLUETOOTH_IRQ_GATTC_INDICATE) {
+        } else if (event == MP_BLUETOOTH_IRQ_GATTC_READ_RESULT) {
+            // conn_handle, value_handle, error, data
+            ringbuf_extract(&o->ringbuf, data_tuple, 2, 1, NULL, 0, NULL, &o->irq_data_data);
```

Comment:
Hm, this inserts a new field into the middle of the tuple, not sure if this is acceptable. It basically breaks `MP_BLUETOOTH_IRQ_GATTC_READ_RESULT` API... @dpgeorge?

I guess reusing `ringbuf_extract` make sense. But maybe we can just swap the items by swapping them in `data_tuple->items`?